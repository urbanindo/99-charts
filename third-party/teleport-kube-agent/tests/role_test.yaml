suite: Role
templates:
  - role.yaml
tests:
  - it: Create a Role when upgrading
    release:
      isupgrade: true
    set:
      unitTestUpgrade: true
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Role

  - it: creates a Role
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Role
      - matchSnapshot: {}
  
  - it: sets Role labels when specified
    values:
      - ../.lint/extra-labels.yaml
    asserts:
      - equal:
          path: metadata.labels.app\.kubernetes\.io/name
          value: teleport-kube-agent
      - equal:
          path: metadata.labels.resource
          value: role
      - matchSnapshot: {}
