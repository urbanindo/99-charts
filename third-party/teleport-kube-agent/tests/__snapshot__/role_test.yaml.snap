creates a Role:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    rules:
    - apiGroups:
      - ""
      resources:
      - secrets
      verbs:
      - create
      - get
      - update
      - patch
sets Role labels when specified:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      labels:
        app.kubernetes.io/name: teleport-kube-agent
        resource: role
      name: RELEASE-NAME
      namespace: NAMESPACE
    rules:
    - apiGroups:
      - ""
      resources:
      - secrets
      verbs:
      - create
      - get
      - update
      - patch
